### ISSUE TITLE
Problems installing ArcGIS Data Store (SendConfigurationApply did not succeed)

### ISSUE THREAD

**Issue by prez13** [Esri/arcgis-powershell-dsc#277]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.0.2

### Affected Resource(s)

* ArcGIS_DataStore

### Configuration Files

Based on single machine base deployment template:

```json
{
    "AllNodes": [
        {
            "NodeName": "azvmage001",
            "DataStoreTypes": [
                "Relational"
            ],
            "Role": [
                "Server",
                "Portal",
                "DataStore",
                "ServerWebAdaptor",
                "PortalWebAdaptor"
            ]
        }
    ],
    "ConfigData": {
        "Version": "10.8",
        "ServerContext": "server",
        "PortalContext": "portal",
        "ServerRole": "GeneralPurposeServer",
        "CloudStorageType": "AzureBlob",
        "Credentials": {
            "ServiceAccount": {
                "UserName": "<removed>",
                "Password": "<removed>",
                "IsDomainAccount": false
            }
        },
        "Server": {
            "LicenseFilePath": "E:\\licences\\ArcGISGISServerAdvanced_ArcGISServer_911838.prvc",
            "Installer": {
                "Path": "E:\\installers\\ArcGIS_Server_Windows_108_172859.exe",
                "InstallDir": "E:\\appl\\ArcGIS\\Server",
                "InstallDirPython": "E:\\appl\\ArcGIS\\Python27"
            },
            "ServerDirectoriesRootLocation": "F:\\arcgisserver\\directories",
            "ConfigStoreLocation": "F:\\arcgisserver\\config-store",
            "LocalRepositoryPath": "F:\\arcgisserver\\local",
            "ServerLogsLocation": "F:\\arcgisserver\\logs",
            "ServerDirectories": [
                {
                    "name": "arcgiscache",
                    "physicalPath": "F:\\arcgisserver\\directories\\arcgiscache",
                    "directoryType": "CACHE",
                    "cleanupMode": "NONE",
                    "maxFileAge": 0
                },
                {
                    "name": "arcgisjobs",
                    "physicalPath": "F:\\arcgisserver\\directories\\arcgisjobs",
                    "directoryType": "JOBS",
                    "cleanupMode": "TIME_ELAPSED_SINCE_LAST_MODIFIED",
                    "maxFileAge": 360
                },
                {
                    "name": "arcgisoutput",
                    "physicalPath": "F:\\arcgisserver\\directories\\arcgisoutput",
                    "directoryType": "OUTPUT",
                    "cleanupMode": "TIME_ELAPSED_SINCE_LAST_MODIFIED",
                    "maxFileAge": 10
                },
                {
                    "name": "arcgissystem",
                    "physicalPath": "F:\\arcgisserver\\directories\\arcgissystem",
                    "directoryType": "SYSTEM",
                    "cleanupMode": "NONE",
                    "maxFileAge": 0
                }
            ],
            "RegisteredDirectories": [
                {
                    "name": "BasemapsCacheDir",
                    "physicalPath": "F:\\arcgisserver\\additionaldirs\\cache",
                    "directoryType": "CACHE"
                },
                {
                    "name": "TestOutputDir",
                    "physicalPath": "F:\\arcgisserver\\additionaldirs\\testOutput",
                    "directoryType": "OUTPUT"
                },
                {
                    "name": "TestJobsDir",
                    "physicalPath": "F:\\arcgisserver\\additionaldirs\\testJobs",
                    "directoryType": "JOBS"
                }
            ],
            "PrimarySiteAdmin": {
                "UserName": "admin",
                "Password": "<removed>"
            }
        },
        "Portal": {
            "LicenseFilePath": "E:\\licences\\ArcGIS_Enterprise_Portal_108_324048_20200624.json",
            "PortalLicenseUserTypeId": "creatorUT",
            "Installer": {
                "Path": "E:\\installers\\Portal_for_ArcGIS_Windows_108_172870.exe",
                "InstallDir": "E:\\appl\\ArcGIS\\Portal",
                "ContentDir": "F:\\arcgisportal"
            },
            "ContentDirectoryLocation":"F:\\arcgisportal\\content",
            "EnableAutomaticAccountCreation": "true",
            "DefaultRoleForUser": "account_publisher",
            "DefaultUserLicenseTypeIdForUser": "creatorUT",
            "PortalAdministrator": {
                "UserName": "admin",
                "Email": "<removed>",
                "Password": "<removed>",
                "SecurityQuestionIndex": 1,
                "SecurityAnswer": "<removed>"
            }
        },
        "DataStore": {
            "ContentDirectoryLocation": "F:\\arcgisdatastore",
            "EnableFailoverOnPrimaryStop": "true",
            "Installer": {
                "Path": "E:\\installers\\ArcGIS_DataStore_Windows_108_172872.exe",
                "InstallDir": "F:\\appl\\ArcGIS\\DataStore"
            }
        },
        "WebAdaptor": {
            "AdminAccessEnabled": true,
            "Installer": {
                "Path": "E:\\installers\\Web_Adaptor_for_Microsoft_IIS_108_172749.exe"
            }
        }
    }
}
```

### Expected Behavior

Install, configure, and licence single machine base deployment.

### Actual Behavior

Got the following error twice (re-run the script once): ```The operation has timed out
PowerShell DSC resource ArcGIS_DataStore  failed to execute Set-TargetResource functionality with error message: Request to https://azvmage001:6443/arcgis/admin/services/System/PublishingTools.GPServer/start failed. Response returned NULL 
The SendConfigurationApply function did not succeed.```

To work around this issue, opened this URL in the browser and started the service manually:
https://azvmage001:6443/arcgis/admin/services/System/PublishingTools.GPServer/start

Then ran the DSC script again, with a new error: ```PowerShell DSC resource ArcGIS_DataStore  failed to execute Set-TargetResource functionality with error message: Register Data Store Failed after multiple attempts. @{code=500; message=274; details=} 
The SendConfigurationApply function did not succeed.```

Attached are the Data Store log files:
[ArcGISDataStore_PS_logs.zip](https://github.com/Esri/arcgis-powershell-dsc/files/4849962/ArcGISDataStore_PS_logs.zip)
And the other ArcGIS PS log files:
[Other_ArcGIS_PS_logs.zip](https://github.com/Esri/arcgis-powershell-dsc/files/4849991/Other_ArcGIS_PS_logs.zip)

### Steps to Reproduce

1. Spin up a Windows 2019 machine on Azure
2. Run this command:  ```Invoke-ArcGISConfiguration -ConfigurationParametersFile "<json file with content mentioned above>" -Mode InstallLicenseConfigure```

### Important Factoids

None.

**Comment by cameronkroeker** [comment]
Hello @prez13,

The module will send a GET request to check the status of the publishingTools prior to configuring Data Store, and in your case this is what was returned:
```
[azvmage001]:[[ArcGIS_DataStore]DataStoreazvmage001] Service Status :- @{configuredState=STARTED; realTimeState=STOPPED}
```

Since the realTimeState=STOPPED, the module then sent a POST request to start publishingtools since its needed for registering Data Store. However, I suspect in this case publishing tools was still in the process of starting which means there was a file lock that wasn't being released. In AGS you may see a message similar to "Another administrative operation is currently accessing the store."

May I ask how many virtual cores (vCPU) the virtual machine has? I recommend trying with at least 4vCPU.

**Comment by cameronkroeker** [comment]
@prez13 

We have included a fix for this issue in v3.1.1.

https://github.com/Esri/arcgis-powershell-dsc/releases/tag/v3.1.1

Closing the issue, however if you still encounter this please feel free to re-open.
